packages/autocomplete-core/src/__tests__/createConcurrentSafePromise.test.ts

```diff
@@ -0,0 +1,41 @@
+import { createConcurrentSafePromise } from '../createConcurrentSafePromise';
+
+function defer<TValue>(fn: () => TValue, timeout: number) {
+  return new Promise<TValue>((resolve) => {
+    setTimeout(() => resolve(fn()), timeout);
+  });
+}
```

Comment:
good strategy for testing these functions, better than the "on-demand-resolving" idea I had